Added test for user name and language completion

dpelle opened the pull request:
This PR adds tests for completion of user names and languages
which were not tested according to codecov:

https://codecov.io/gh/vim/vim/src/33c5e9fa7af935c61a8aac461b9664c501003440/src/misc1.c#L4627
https://codecov.io/gh/vim/vim/src/33c5e9fa7af935c61a8aac461b9664c501003440/src/ex_cmds2.c#L5634

dpelle committed:
Added test for user name and language completion

dpelle committed:
Skip tests of user names and languages on Windows

Can we make these test pass on Windows?

dpelle commented:
I skipped the newly added tests for Windows, as I'm not sure why user name and
language completion do not work on Windows. It would be good if someone who
has Windows could enable and fix Test_cmdline_complete_user_names() and
Test_cmdline_complete_languages() for Windows.

brammool commented:
Dominique wrote:

> This PR adds tests for completion of user names and languages
> which were not tested according to codecov:

Thanks.  Please mark the missing parts for Windows with "TODO", so that
these are easy to find.


-- 
hundred-and-one symptoms of being an internet addict:
23. You can't call your mother...she doesn't have a modem.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

dpelle committed:
Missing TODO in tests skipped for Windows

As requested in this comment:
  https://github.com/vim/vim/pull/2978#issuecomment-394111428

k-takata commented on src/testdir/test_cmdline.vim:
```diff
@@ -392,6 +391,42 @@ func Test_cmdline_complete_user_cmd()
 
+func Test_cmdline_complete_user_names()
+  if has('unix') && executable('whoami')
+    " TODO: can wk make this test work on Windows?
```
typo? wk -> we

dpelle replied:
Thanks @k-takata. I pushed a change to update the comment.
It now mentions the entry in todo.txt:

> Patch to support user name completion on MS-Windows. (Yasuhiro
> Matsumoto, 2012 Aug 16)

k-takata commented:
User name completion on Windows is in the todo list:

> Patch to support user name completion on MS-Windows. (Yasuhiro Matsumoto, 2012
> Aug 16)

Language completion is not implemented.

dpelle committed:
Mention that user completion is not implemented on Windows
